zebra/zebra_dplane.c

```diff
@@ -3706,21 +3866,36 @@ enum zebra_dplane_result dplane_neigh_discover(const struct interface *ifp,
  * Common helper api for neighbor updates
  */
 static enum zebra_dplane_result
-neigh_update_internal(enum dplane_op_e op,
-		      const struct interface *ifp,
-		      const struct ethaddr *mac,
-		      const struct ipaddr *ip,
-		      uint32_t flags, uint16_t state,
-			  uint32_t update_flags)
+neigh_update_internal(enum dplane_op_e op, const struct interface *ifp,
+		      const void *link, const int link_family,
+		      const struct ipaddr *ip, uint32_t flags, uint16_t state,
+		      uint32_t update_flags, int protocol)
```

Comment:
you've added 'protocol' , but it's not used? since you expect to be able to use the 'type' in the netlink module, that seems wrong?

for the vtep/evpn/vxlan path, I think you have to find a way to get RTPROT_ZEBRA into the netlink buffer without the unprotected debug in `zebra2proto()` going off for every update. just passing a zero through doesn't seem correct.

Reply:
missing `dplane_ctx_set_type(ctx, protocol);`.
I created a new route type called `ZEBRA_ROUTE_NEIGH` that avoids calling unprotected debug.